crates/forge/src/lib.rs

```diff
@@ -462,35 +487,101 @@ fn run_with_fuzzing(
     };
     let mut fuzzer = RandomFuzzer::create(fuzzer_seed, fuzzer_runs, &args)?;
 
-    let mut results = vec![];
+    let mut tasks = FuturesUnordered::new();
+
+    let (send, mut recv) = channel(1);
 
     for _ in 1..=fuzzer_runs {
         let args = fuzzer.next_args();
+        let case = case.clone();
+        let case_copy = case.clone();
+        let runner = runner.clone();
+        let args = args.clone();
+        let cancellation_fuzzing_token = token.clone();
+        let runner_config = runner_config.clone();
+        let send = send.clone();
+
+        tasks.push(task::spawn(async move {
+            tokio::select! {
+                () = runner_config.error_cancellation_token.cancelled() => {
+                    //Stop executing all tests because flag --exit-first'
+                    //has been set and one test FAIL
+                    Ok(TestCaseSummary::None {  })
+                },
+                () = runner_config.cancellation_token.cancelled() => {
+                    //Stop executing all tests because flag --exit-first'
+                    //has been set and one test FAIL
+                    Ok(TestCaseSummary::skipped(&case_copy))
+                },
+                () = cancellation_fuzzing_token.cancelled() => {
+                    //Stop executing all tests because flag --exit-first'
+                    //has been set and one test FAIL
+                    Ok(TestCaseSummary::skipped(&case_copy))
+                },
```

Comment:
This description is missleading, it is not cancelled bcs of `exit_first`, but because of fuzzing not being run further after the first failed run. 

Also I would introduce a `TestCaseSummary::SkippedDuringFuzzing` or sth like that. This ofc works, but the test is not actually skipped in the sense of `exit_first` flag

Reply:
Thx for the catch! I  overlooked it! `cancellation_fuzzing_token.cancelled()` is executed only if signle fuzzing test fails. So the user will get `TestCaseSummary::Failed` after the use of this token!